### DISCUSSION TITLE
User Tag limits

### DISCUSSION THREAD

**Discussion by JeltevanBoheemen** [UUDigitalHumanitieslab/I-analyzer#1579] [General]
How do we want to enforce a sensible amount of tags?
Suggestion by @lukavdplas:
It could make more sense to impose the limit on a (User, Corpus) scope, rather than (User, Tag)

**Comment by lukavdplas** [comment]
To summarise my comments on this: it depends on why you want to limit tags in the first place.

Since tags are private for logged-in users, spamming should not be an issue. The memory used by tags is tiny compared to the rest of the application, so I also see no reason to limit that.

My argument for limiting tags at all would be that, in our current implementation, the way we implement searching (#1148) is to make a query with a filter that  lists all document IDs with that tag. So there is a point where the payload of the request too large.

(Corpus, User) scope makes sense if we assume that:

- users can search for the union of several tags
- users cannot search through multiple corpora

But since we do have plans to implement cross-corpus search (#992), perhaps the scope should not be limited by the corpus.

So then we should use only the user as the scope, I guess?

*However*, I could imagine that large tags can be filtered individually, but not combined when searching. (So once you select some tags in the tag filter, other tags may be disabled.)
